src/NuGet.Core/NuGet.Common/Errors/ILogFileContext.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace NuGet.Common
+{
+    public interface ILogFileContext
+    {
+        /// <summary>
+        /// Indicates the file for which the error was thrown.
+        /// </summary>
+        string FilePath { get; set; }
```

Comment:
``set;`` should be removed from all of the LogMessage interfaces. Callers should only need to get the data.

For scenarios where we need to write to the messages the concrete type should be used.